gulpfile.js

```diff
@@ -210,9 +213,9 @@ gulp.task("watch", ["serve"], function () {
     );
     gulp.watch(
         [
-            path.join(ROOT_DIR, "**", "*.yml"),
-            path.join(SOURCE_DIR, "**", "*.html"),
-            path.join(SOURCE_DIR, "**", "*.md"),
+            path.join(DATA_DIR, "**", "*.yml"),
+            path.join(SOURCE_DIR, "**", "*.html") + "!" + path.join(DOCS_DIR, "**"),
+            path.join(SOURCE_DIR, "**", "*.md") + "!" + path.join(DOCS_DIR, "**") ,
```

Comment:
This won't regenerate the site if the docs change. Perhaps wrap this in `if (gutil.env.nodocs)`.